evalml/model_understanding/graphs.py

```diff
@@ -436,6 +436,12 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
     elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
         pipeline._estimator_type = "regressor"
     pipeline.feature_importances_ = pipeline.feature_importance
+    if isinstance(feature, int):
+        if X.iloc[:, feature].isnull().sum():
+            warnings.warn("There are null values in the features, which will cause NaN values in the partial dependency. Fill in these values to remove the NaN values.")
+    if isinstance(feature, str):
+        if X[feature].isnull().sum():
+            warnings.warn("There are null values in the features, which will cause NaN values in the partial dependency. Fill in these values to remove the NaN values.")
```

Comment:
Looks good! A few nit-picks:
1. Please declare a warning type! We don't have this pattern in place yet, but: add a [subclass `UserWarning`](https://docs.python.org/3/library/exceptions.html#UserWarning) in `evalml/exceptions/exceptions.py`, add a docstring there, and then include that class here in the call to `warn`, after the message. Why do this: allows users to apply the `warnings` filters [based on category](https://docs.python.org/3/library/warnings.html#overriding-the-default-filter) as well as message contents. We just ran into this same consideration with the text / datatables work, in fact!
2. "NaN values in the partial dependency" --> "NaN values in the partial dependence output" ?
3. style:
```python
if ((isinstance(feature, int) and X.iloc[:, feature].isnull().sum()) or
    (isinstance(feature, str) and X[feature].isnull().sum())):
    warnings.warn(...)
```